### PR TITLE
Document release managers with initial schedule

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

This is based on the SIG discussion https://docs.google.com/document/d/1Unbs2qp_j5kp8FfL_lRH-ld7i5EOQpsq0I4djkOOSL4/edit

@jpkrohling @VineethReddy02 @yuriolisa @frzifus @jaronoff97  please review.

### CODE REVIEW

**Review by frzifus — APPROVED** [review]

File: RELEASE.md
Comment (jaronoff97) [inline RELEASE.md:36]:
```suggestion
The operator should be released within a week after the [OpenTelemetry collector release](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md#release-schedule).
```

**Review by jaronoff97 — COMMENTED** [review]
One nit, otherwise looks good!

**Review by yuriolisa — APPROVED** [review]

File: RELEASE.md
Comment (jpkrohling) [inline RELEASE.md:41]: I guess it's my time now to step out from maintainership. I was already away for a while, just lurking around in case something needed someone to break a deadlock, but there are enough maintainers now so this problem won't happen. I also trust that the current maintainers are doing a great job, and I don't deserve to be called a "maintainer" and be compared with you, the ones actually doing the work.

Comment (pavolloffay) [reply RELEASE.md:41]: Thanks for bootstrapping the project and all the work. 

I can wait for your PR if you prefer.

Comment (jaronoff97) [reply RELEASE.md:41]: Thank you @jpkrohling ❤️ @pavolloffay If you'd like, you can change his name for mine.

File: RELEASE.md
Comment (frzifus) [inline RELEASE.md:44]:
```suggestion
| v0.69.0 | @pavolloffay    |
| v0.70.0 | @VineethReddy02 |
| v0.71.0 | @frzifus        |
| v0.72.0 | @yuriolisa      |
```

**Review by frzifus — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

File: RELEASE.md
Comment (frzifus) [reply RELEASE.md]: > Thank you @jpkrohling heart @pavolloffay If you'd like, you can change his name for mine.

Ups, let me revert what i did :roll_eyes: :x:

File: RELEASE.md
Comment (frzifus) [inline RELEASE.md:43]:
```suggestion
| v0.72.0 | @yuriolisa      |
| v0.73.0 | @jaronoff97     |
```

Comment (frzifus) [reply RELEASE.md:43]: @jaronoff97 added you at the end ^^

**Comment by pavolloffay** [comment]
We didn't get approval from @VineethReddy02. I will reach out to him directly on Slack

**Comment by VineethReddy02** [comment]
LGTM! would love to be a release manager more often! :)

**Pull request merged**
